src/auth.js

```diff
@@ -131,12 +135,14 @@ export async function authorizeEntry(
   const payload = hash(preimage.toXDR());
 
   let signature;
+  let publicKey;
   if (typeof signer === 'function') {
     signature = Buffer.from(await signer(preimage));
+    publicKey = Address.fromScAddress(addrAuth.address()).toString();
```

Comment:
Hi @kalepail, I think we can make the `SigningCallback` return both the `signature` and the `public_key` at the same time, which will give users more flexibility. What do you think?

Reply:
This would be a breaking change to the API but it seems like a good idea to solve this issue for both cases! We can include this in the protocol 22 release.